kraken-build/src/kraken/std/python/buildsystem/uv.py

```diff
@@ -62,136 +63,143 @@ def unsafe_url(self) -> str:
-        for index in self.supplemental:
-            args += ["--extra-index-url", index.safe_url]
+        args: list[str] = []
+        for index in self.indexes:
+            args += ["--default-index" if index.default else "--index", index.safe_url]
         return args
 
     def to_unsafe_args(self) -> list[str]:
         """Create a list of arguments for UV with sensitive information in plaintext."""
 
-        args = []
-        if self.primary is not None:
-            args += ["--index-url", self.primary.unsafe_url]
-        for index in self.supplemental:
-            args += ["--extra-index-url", index.unsafe_url]
+        args: list[str] = []
+        for index in self.indexes:
+            args += ["--default-index" if index.default else "--index", index.unsafe_url]
         return args
 
-    def to_config(self, config: MutableMapping[str, Any]) -> None:
+    def to_config(self) -> list[dict[str, Any]]:
         """Inject UV configuration for indexes into a configuration."""
-
-        if self.primary is None:
-            config.pop("index-url", None)
-        else:
-            config["index-url"] = self.primary.url
-
-        if not self.supplemental:
-            config.pop("extra-index-url", None)
-        else:
-            config["extra-index-url"] = [idx.url for idx in self.supplemental]
+        entries: list[dict[str, str | bool]] = []
+        for index in self.indexes:
+            entry: dict[str, Any] = {"url": index.url}
+            if len(index.name) > 0:
+                entry["name"] = index.name
+            if index.default:
+                entry["default"] = True
+            if index.explicit:
+                entry["explicit"] = True
+            entries.append(entry)
+        return entries
 
     def to_env(self) -> dict[str, str]:
+        """Convert UV configuration for indexes into a environment variables."""
+
         env = {}
-        if self.primary is not None:
-            env["UV_INDEX_URL"] = self.primary.unsafe_url
-        if self.supplemental:
-            env["UV_EXTRA_INDEX_URL"] = os.pathsep.join(idx.unsafe_url for idx in self.supplemental)
+        uv_indexes = []
+        for index in self.indexes:
+            if index.default:
+                # https://docs.astral.sh/uv/configuration/environment/#uv_default_index
+                env["UV_DEFAULT_INDEX"] = index.unsafe_url
+            else:
+                # https://docs.astral.sh/uv/configuration/environment/#uv_index
+                uv_indexes.append((f"{index.name}=" if index.name != "" else "") + index.unsafe_url)
+
+        if len(uv_indexes) != 0:
+            env["UV_INDEX"] = " ".join(uv_indexes)
         return env
 
 
 class UvPyprojectHandler(PyprojectHandler):
     """Implements the PyprojectHandler interface for UV projects."""
 
-    # TODO: Support `uv.toml` configuration file?
-
-    # PyprojectHandler
+    # TODO: Support global `uv.toml` configuration file?
```

Comment:
I'd avoid something as system dependent as "global", but supporting an "external" `uv.toml` provided by its path might be nice.

Reply:
I agree!